pkg/http-handler/http-handler.go

```diff
@@ -261,17 +262,67 @@ func DeleteAppRepository(kubeHandler kube.AuthHandler) func(w http.ResponseWrite
 
 		clientset, err := kubeHandler.AsUser(token, requestCluster)
 		if err != nil {
-			returnK8sError(err, w)
+			returnK8sError(err, "delete", "AppRepository", w)
 			return
 		}
 
 		err = clientset.DeleteAppRepository(repoName, requestNamespace)
 		if err != nil {
-			returnK8sError(err, w)
+			returnK8sError(err, "delete", "AppRepository", w)
 		}
 	}
 }
 
+// GetAppRepository gets an App Repository with a related secret if present.
+func GetAppRepository(kubeHandler kube.AuthHandler) func(w http.ResponseWriter, req *http.Request) {
```

Comment:
I guess this function will get eventually moved to the new repositories API

Reply:
Yep, indeed, but that needs to be properly designed for a more generic approach etc.